ci/test_docker.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -e
+# Test Snowflake SQLAlchemy in Docker
+# NOTES:
+#   - By default this script runs Python 3.7 tests, as these are installed in dev vms
+#   - To compile only a specific version(s) pass in versions like: `./test_docker.sh "3.7 3.8"`
+
+set -o pipefail
+
+# In case this is ran from dev-vm
+PYTHON_ENV=${1:-3.7}
+
+# Set constants
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+SQLALCHEMY_DIR="$( dirname "${THIS_DIR}")"
+WORKSPACE=${WORKSPACE:-$SQLALCHEMY_DIR}
+source $THIS_DIR/set_base_image.sh
+
+cd $THIS_DIR/docker/sqlalchemy_build
+
+CONTAINER_NAME=test_sqlalchemy
+
+echo "[Info] Building docker image"
+arch=$(uname -p)
+
+echo "[Info] Building docker image"
+if [[ "$arch" == "aarch64" ]]; then
+  BASE_IMAGE=$BASE_IMAGE_MANYLINUX2014AARCH64
+  GOSU_URL=https://github.com/tianon/gosu/releases/download/1.14/gosu-arm64
+else
+  BASE_IMAGE=$BASE_IMAGE_MANYLINUX2014
+  GOSU_URL=https://github.com/tianon/gosu/releases/download/1.14/gosu-amd64
+fi
+
+echo "[Info] Start building docker image"
+docker build --pull -t ${CONTAINER_NAME}:1.0 --build-arg BASE_IMAGE=$BASE_IMAGE  --build-arg GOSU_URL="$GOSU_URL" . -f Dockerfile
```

Comment:
Very recently I found a better way to build and consume Docker images temporarily. Please remove this line and modify the docker run command similar to https://github.com/snowflakedb/snowflake-sqlalchemy/blob/5d17bfb3dbfb1a9b29d1156c1da538ecf61847e9/ci/build_docker.sh#L24

This way the docker image is removed after the run command finishes and the built layers are not kept around.

Reply:
nice finding, updated